cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,65 @@
-/// <reference types="cypress" />
-/// <reference types="../support" />
+/// <reference types='cypress' />
+/// <reference types='../support' />
+
+import SignInPageObject from '../support/pages/signIn.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+import UserSettings from '../support/pages/userSettings.pageObject';
+
+const signInPage = new SignInPageObject();
+const homePage = new HomePageObject();
+const userSettings = new UserSettings();
 
 describe('Settings page', () => {
+  let user;
   before(() => {
-
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
   beforeEach(() => {
-
+    cy.task('db:clear');
+    cy.login();
+    userSettings.visit();
   });
 
   it('should provide an ability to update username', () => {
-
+    userSettings.typeUsername(user.username);
+    userSettings.clickUpdateSettingsBtn();
+    homePage.assertHeaderContainUsername(user.username);
   });
 
   it('should provide an ability to update bio', () => {
-
+    userSettings.typeBio(user.bio);
+    userSettings.clickUpdateSettingsBtn();
+    userSettings.assertBioUpdated(user.bio);
   });
 
   it('should provide an ability to update an email', () => {
-
+    userSettings.typeEmail(user.email);
+    userSettings.clickUpdateSettingsBtn();
+    userSettings.visit();
+    userSettings.clickLogoutBtn();
+    signInPage.visit();
+    signInPage.typeEmail(user.email);
+    signInPage.typePassword(user.password);
+    signInPage.clickSignInBtn();
+    homePage.assertHeaderContainUsername('riot');
```

Comment:
The assertion `homePage.assertHeaderContainUsername('riot');` seems incorrect. It should assert the username of the logged-in user, which is `user.username`, instead of the hardcoded value 'riot'.